src/chainstate/stacks/db/transactions.rs

```diff
@@ -1153,12 +1155,18 @@ impl StacksChainState {
                     .expect("BUG: total block cost decreased");
                 let sponsor = tx.sponsor_address().map(|a| a.to_account_principal());
 
+                // Compile the contract to Wasm
+                let mut module =
+                    compile_contract(&mut contract_analysis).expect("Failed to compile contract");
```

Comment:
We're probably going to want an actual error here if the contract fails to compile to WASM instead of panicing

Reply:
Addressed in 291c4a555